leaphymicropython/actuators/dcmotor.py

```diff
@@ -1,117 +1,96 @@
+        self.direction.value(1)
+        self.validate_speed(speed)
+        speed_u16 = self.convert_speed_to_duty_u16(speed)
+        self.pwm.duty_u16(speed_u16)
 
     def backward(self, speed: int):
         """
-        Moves the motor backward at the specified speed.
-
-        :param speed: The speed value for backward motion.
+        Sets the DC motor to backward
+        :param speed: int, the speed of the motor
         """
-        self.convert_ccw(speed)
-        set_pwm(self.motor_a, self.value_speed_ccw, 20000)
-        set_pwm(self.motor_b, self.value_speed_ccw, 20000)
-        set_pin(self.enable_a, True)
-        set_pin(self.enable_b, True)
+        self.direction.value(0)
+        self.validate_speed(speed)
+        speed_u16 = self.convert_speed_to_duty_u16(speed)
+        self.pwm.duty_u16(speed_u16)
 
-    def left(self, speed: int, direction: int):
+    def stop(self):
         """
-        Turns the motor left at the specified speed and direction.
-
-        :param speed: The speed value for turning.
-        :param direction: The direction for turning (1 for CW, 0 for CCW).
+        Stops the DC motor
         """
-        if direction == 1:
-            self.convert_cw(speed)
-            set_pwm(self.motor_a, self.value_speed_cw, 20000)
-            set_pwm(self.motor_b, self.value_speed_cw / 8, 20000)
-            set_pin(self.enable_a, True)
-            set_pin(self.enable_b, True)
-        elif direction == 0:
-            self.convert_ccw(speed)
-            set_pwm(self.motor_a, self.value_speed_cw, 20000)
-            set_pwm(self.motor_b, self.value_speed_cw / 8, 20000)
-            set_pin(self.enable_a, True)
-            set_pin(self.enable_b, True)
+        self.pwm.duty_u16(0)
 
-    def right(self, speed: int, direction: int):
+    def test(self):
         """
-        Turns the motor right at the specified speed and direction.
+        Tests the DC motor
+        """
+        self.forward(100)
+        sleep(1)
+        self.backward(100)
+        sleep(1)
+        self.stop()
+        sleep(1)
 
-        :param speed: The speed value for turning.
-        :param direction: The direction for turning (1 for CW, 0 for CCW).
+    def validate_speed(self, speed):
+        """
+        Validates the speed of the motor
+        :param speed: int, the speed of the motor
         """
-        if direction == 1:
-            self.convert_cw(speed)
-            set_pwm(self.motor_a, self.value_speed_cw / 8, 20000)
-            set_pwm(self.motor_b, self.value_speed_cw, 20000)
-            set_pin(self.enable_a, True)
-            set_pin(self.enable_b, True)
-        elif direction == 0:
-            self.convert_ccw(speed)
-            set_pwm(self.motor_a, self.value_speed_cw / 8, 20000)
-            set_pwm(self.motor_b, self.value_speed_cw, 20000)
-            set_pin(self.enable_a, True)
-            set_pin(self.enable_b, True)
+        assert 0 <= speed <= 100, "Speed must be between 0 and 100"
```

Comment:
Assert is niet bedoeld voor prod code: beter om een 'raise ValueError' te gebruiken